consensus/src/dag/dag_driver.rs

```diff
@@ -356,10 +360,37 @@ impl RpcHandler for DagDriver {
     }
 }
 
-impl Drop for DagDriver {
-    fn drop(&mut self) {
-        if let Some((handle, _)) = self.rb_abort_handle.lock().as_ref() {
-            handle.abort()
+struct BoundedVecDeque<T> {
```

Comment:
nit: probably move it to a util, and use it for the sliding window inside MetadataBackendAdapter too. also small unit tests would be good